mlir/include/mlir/Dialect/Rock/IR/RockOps.td

```diff
@@ -1145,6 +1145,46 @@ def Rock_AccelGemmOp:
   let hasVerifier = 1;
 }
 
+// blockwise_broadcasting_reduction
+def Rock_BlockwiseBroadcastReduceOp:
+    Rock_Op<"blockwise_broadcast_reduce">,
+    Arguments<(ins MemRefOf<GemmOutputTypes>:$input,
+                   MemRefOf<LdsBufferTypes>:$workspace_buffer,
+                   MemRefOf<GemmOutputTypes>:$output,
+                   IndexAttr:$axis,
+                   ReduceMethodAttr:$reduceMethod,
+                   TransformMapArrayAttr:$inputRegView,
+                   I32Attr:$blockSize
+                   )> {
+  let summary = "Blockwise reduce operation with output broadcasting";
+  let description = [{
+    The `rock.blockwise_broadcast_reduce` op does reduce at a blockwise level and
+    broadcast the reduce values to a size of input tensor. Note that both input/output
+    tensors live in registers.
+
+    $input tensor resides in registers in threads.
+
+    $inputRegView should be register {tid, iter}
+    to virtual tensor {d0, ... , Dr , ... , dn} which is a slice of
+    a larger tensor {D0, ... , Dr , ... , Dn}. Note that the reduction
+    axis (Dr) is not sliced because it is not supported.
+
+    $output tensor resides in registers in threads. It would have
+    the same dimensionality of the input tensor but broadcasted.
+    So the users of the op, has to add a extraView if they are writing back
+    to some memory (LDS/Global) to bar threads (or iteration within threads)
+    to not to create a broadcasted tensor.
+
+    $workspace_buffer resides in LDS. This should be a flat tensor with at least
+    with number of items per block.
```

Comment:
with at least `blockSize` elements.

Reply:
hmm no, a single thread could have multiple items to reduce with other items in other threads. I reworded with "with at least number of items to reduce per block."